### ISSUE TITLE
OneOf errors when struct contains pointer types

### ISSUE THREAD

**Issue by williammartin** [discovery-digital/jsonschema#21]
```
package main

import (
	"encoding/json"
	"fmt"
	"reflect"

	"github.com/discovery-digital/jsonschema"
)

type Foo struct {
	Bar string `json:"bar"`
	Baz int    `json:"baz"`
}

func (f *Foo) OneOf() []reflect.StructField {
	bar, _ := reflect.TypeOf(f).FieldByName("Bar")
	baz, _ := reflect.TypeOf(f).FieldByName("Baz")
	return []reflect.StructField{
		bar,
		baz,
	}
}

func main() {
	reflector := &jsonschema.Reflector{}
	js := reflector.Reflect(&Foo{})
	jm, err := json.Marshal(js)

	if err != nil {
		fmt.Println(err)
	}

	fmt.Println(string(jm))
}
```

In this example program, because we pass in a pointer to `Foo` at reflection time, we get an error like:

```
→ go run main.go
panic: interface conversion: **main.Foo is not jsonschema.oneOf: missing method OneOf

goroutine 1 [running]:
github.com/discovery-digital/jsonschema.(*Reflector).reflectTypeToSchema(0xc00005ff70, 0xc000096510, 0x111dee0, 0x10e2500, 0x0)
        /Users/wmartin/go/src/github.com/discovery-digital/jsonschema/reflect.go:245 +0x449
github.com/discovery-digital/jsonschema.(*Reflector).ReflectFromType(0xc00005ff70, 0x111dee0, 0x10e2500, 0x10ea0c0)
        /Users/wmartin/go/src/github.com/discovery-digital/jsonschema/reflect.go:148 +0x2ff
github.com/discovery-digital/jsonschema.(*Reflector).Reflect(0xc00005ff70, 0x10e2500, 0xc00000c1c0, 0x0)
        /Users/wmartin/go/src/github.com/discovery-digital/jsonschema/reflect.go:128 +0x5a
main.main()
        /Users/wmartin/go/src/github.com/williammartin/scratchpad/main.go:27 +0x9d
exit status 2
```

In the simple case this is trivial to fix by passing in a value type, but I have already defined structs that contain pointer within the tree, so it's not really viable for me to change that.

**Comment by welps** [comment]
@williammartin I honestly didn't realize this was being used externally until I saw a PR last week. I've addressed the bug you've brought up in #22 . Thanks for reporting and sorry for the delay.

**Comment by williammartin** [comment]
hehe no problem. That's the beauty of OSS. I just followed the breadcrumbs to this fork which it looked like you were semi maintaining and had some stuff I wanted :)
